src/app/duration-picker/duration-picker.component.ts

```diff
@@ -156,74 +203,106 @@ export class DurationPickerComponent implements OnInit, ControlValueAccessor {
   }
 
-  generate(): string {
-    let output = 'P';
+  generate(): string | number {
+    if (this.mode === 'ISO_8601') {
+      let output = 'P';
 
-    if (this.config.showNegative && this.negative) {
-      output = '-' + output;
-    }
+      if (this.config.showNegative && this.negative) {
+        output = '-' + output;
+      }
 
-    if (this.config.showYears && this.years) {
-      output += `${this.years}Y`;
-    }
-    if (this.config.showMonths && this.months) {
-      output += `${this.months}M`;
-    }
-    if (this.config.showWeeks && this.weeks) {
-      output += `${this.weeks}W`;
-    }
-    if (this.config.showDays && this.days) {
-      output += `${this.days}D`;
-    }
-    if (
-      (this.config.showHours && this.hours)
-      || (this.config.showMinutes && this.minutes)
-      || (this.config.showSeconds && this.seconds)
-    ) {
-      output += 'T';
-
-      if (this.config.showHours && this.hours) {
-        output += `${this.hours}H`;
+      if (this.config.showYears && this.years) {
+        output += `${this.years}Y`;
       }
-      if (this.config.showMinutes && this.minutes) {
-        output += `${this.minutes}M`;
+      if (this.config.showMonths && this.months) {
+        output += `${this.months}M`;
       }
-      if (this.config.showSeconds && this.seconds) {
-        output += `${this.seconds}S`;
+      if (this.config.showWeeks && this.weeks) {
+        output += `${this.weeks}W`;
+      }
+      if (this.config.showDays && this.days) {
+        output += `${this.days}D`;
+      }
+      if (
+        (this.config.showHours && this.hours)
+        || (this.config.showMinutes && this.minutes)
+        || (this.config.showSeconds && this.seconds)
+      ) {
+        output += 'T';
+
+        if (this.config.showHours && this.hours) {
+          output += `${this.hours}H`;
+        }
+        if (this.config.showMinutes && this.minutes) {
+          output += `${this.minutes}M`;
+        }
+        if (this.config.showSeconds && this.seconds) {
+          output += `${this.seconds}S`;
+        }
       }
-    }
 
-    // if all values are empty, just output null
-    if (output === 'P' || output === '-P') {
-      output = this.config.zeroValue;
-    }
+      // if all values are empty, just output null
+      if (output === 'P' || output === '-P') {
+        output = this.config.zeroValue;
+      }
 
-    return output;
+      return output;
+    } else {
+      const currentValueInSeconds = (this.years * DURATION_UNITS.years) +
+        (this.months * DURATION_UNITS.months) +
+        (this.weeks * DURATION_UNITS.weeks) +
+        (this.days * DURATION_UNITS.days) +
+        (this.hours * DURATION_UNITS.hours) +
+        (this.minutes * DURATION_UNITS.minutes) +
+        (this.seconds * DURATION_UNITS.seconds);
+      const duration = Math.floor(currentValueInSeconds / DURATION_UNITS[this.mode]);
+      return this.config.showNegative && this.negative ? (duration * -1) : duration;
+    }
```

Comment:
I think it's better to split the generation of different modes in different methods, then `generate()` would just call something like `generateISO()` or `generateNumeric()`. Also please do not change the indentation style.